### PR TITLE
feat(useDateFormat): can't return zero value

### PR DESCRIPTION
**Pull request by JocelynFloresz**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description
fix https://github.com/vueuse/vueuse/issues/3268. when return data is 0, it should be the last value of || operator

### Additional context

---

### <samp>🤖 Generated by Copilot at e9403c1</samp>

Refactored and improved `useDateFormat` function by simplifying `formatDate` logic and fixing a potential bug. Updated `packages/shared/useDateFormat/index.ts` file.

### <samp>🤖 Generated by Copilot at e9403c1</samp>

* Refactored and improved `useDateFormat` function ([link](https://github.com/vueuse/vueuse/pull/3270/files?diff=unified&w=0#diff-314bd9687a1ae662b93b564d3b955d5dd76051e4a97adcf2a12e842e913b3970L69-R69),

### CODE REVIEW

File: packages/shared/useDateFormat/index.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export function formatDate(date: Date, formatStr: string, options: UseDateFormat
     a: () => meridiem(hours, minutes, true),
     aa: () => meridiem(hours, minutes, true, true),
   }
-  return formatStr.replace(REGEX_FORMAT, (match, $1) => $1 || matches[match]?.() || match)
+  return formatStr.replace(REGEX_FORMAT, (match, $1) => $1 || matches[match]?.())
```
Comment (antfu) [inline packages/shared/useDateFormat/index.ts:69]: We should use `??` instead

Comment (JocelynFloresz) [reply packages/shared/useDateFormat/index.ts:69]: should I change by this way? `matches[match]??()`.

Comment (antfu) [reply packages/shared/useDateFormat/index.ts:69]: I mean 

```ts
 return formatStr.replace(REGEX_FORMAT, (match, $1) => $1 ?? matches[match]?.() ?? match)
```

Comment (JocelynFloresz) [reply packages/shared/useDateFormat/index.ts]: ok! Wait me a moment please, this is my first PR on GitHub.

**Pull request closed without merging**